lxd/storage/drivers/interface.go

```diff
@@ -0,0 +1,52 @@
+package drivers
+
+import (
+	"github.com/lxc/lxd/lxd/migration"
+	"github.com/lxc/lxd/lxd/operations"
+	"github.com/lxc/lxd/shared/api"
+)
+
+// VolumeType represents a storage volume type.
+type VolumeType string
+
+// VolumeTypeCustom represents a custom storage volume.
+const VolumeTypeCustom = VolumeType("custom")
```

Comment:
@stgraber there are similar typea already defined in the db package, such as `StoragePoolVolumeTypeNameContainer` that is "container". There is a conversion function that converts from these string types to the DB int types, however as we are now using `VolumeTypeContainer` type this cannot be used with the existing conversion function (which I need to lookup the volume ID).

I can create another conversion function that takes `VolumeType` and converts to DB int, or perhaps we could use the existing db string constants instead, and then we can just append an "s" to the parts where its needed for the storage paths. What do you think?

Reply:
Except for the part where the plural of `custom` is still `custom` in our case :)

I think we may need a `volumetype` package which gives us a struct for all of:
 - custom
 - custom snapshot
 - container
 - container snapshot
 - image
 - virtual machine
 - virtual machine snapshot

With the DB type for each of them, whether it's a snapshot or not, the API name, on-disk name, ... And port everything to using that.

Reply:
@stgraber OK I'll keep that under advisement, for now I've avoided the need for another conversion function because after I found out that there was not a need to support copying different types of volumes to one and other, I was then able to hard code the new volume types based on the context of the function, `CreateCustomVolumeFromCopy` for example.